relayer/src/chain/cosmos.rs

```diff
@@ -373,13 +383,64 @@ impl CosmosSdkChain {
         let mut tx_bytes = Vec::new();
         prost::Message::encode(&tx_raw, &mut tx_bytes).unwrap();
 
-        let response = self.block_on(broadcast_tx_sync(self, tx_bytes))?;
+        self.block_on(broadcast_tx_sync(self, tx_bytes))
+    }
+
+    // Try to send_tx with retry on account sequence error.
+    // An account sequence error can occur if the cached account sequence of
+    // the relayer somehow get outdated, or when the relayer's wallet is used
+    // concurrently elsewhere.
+    // When there is an account sequence mismatch, we refetch the account sequence
+    // from the chain and retry sending the transaction again with the new sequence.
+    fn send_tx_with_account_sequence_retry(
+        &mut self,
+        proto_msgs: Vec<Any>,
+        retries: u32,
+    ) -> Result<Response, Error> {
+        let account_sequence = self.account_sequence()?;
+
+        let response = self.send_tx_with_account_sequence(proto_msgs.clone(), account_sequence)?;
 
         debug!("[{}] send_tx: broadcast_tx_sync: {:?}", self.id(), response);
 
-        self.incr_account_sequence()?;
+        match response.code {
+            Code::Ok => {
+                self.incr_account_sequence();
+                Ok(response)
+            }
+            Code::Err(code) if code == INCORRECT_ACCOUNT_SEQUENCE_ERR => {
```

Comment:
You can use constants in pattern matches:

```suggestion
            Code::Err(INCORRECT_ACCOUNT_SEQUENCE_ERR) => {
```

Reply:
I tried before and it didn't work. Rust does not allow pattern matching with constants.

Reply:
Really? It works for me, at least in some capacity, see: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=2bfcd350cf71b52e4954a8c127a1784d

Reply:
Just tried in the actual code and it works as well :)

Reply:
Ok. Probably it was because I mistyped the constant as `u64` instead of `u32` before, and then got mistaken of the actual error message when I tried to do inline constant pattern matching. Fixed it now.